version: "3"

networks:
  web:
    external: true
  internal:
    external: false

services:
  cells:
    image: pydio/cells:latest
    restart: unless-stopped
    ports: [ "8080:8080" ]
    environment:
      - CELLS_LOG_LEVEL=production
      - CELLS_BIND=:8080
      - CELLS_EXTERNAL=https://drive.ovcloud.io
      - CELLS_NO_TLS=1
    volumes:
      - data:/var/cells/data
      - cellsdir:/var/cells
    labels:
      - traefik.http.routers.drive.rule=Host(`drive.ovcloud.io`)
      - traefik.http.routers.drive.tls=true
      - traefik.http.routers.drive.tls.certresolver=lets-encrypt
      - traefik.port=80
    networks:
      - internal
      - web
    depends_on:
      - db

  db:
    image: mysql:8
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: password
      MYSQL_DATABASE: cells
      MYSQL_USER: cells
      MYSQL_PASSWORD: password
    networks:
      - internal
    labels:
      - traefik.enable=false

  collabora:
    image: collabora/code:latest
    restart: unless-stopped
    networks:
      - internal
    labels:
      - traefik.enable=false

  phpmyadmin:
    image: phpmyadmin
    restart: always
    environment:
      - PMA_ARBITRARY=1
    labels:
      - traefik.http.routers.adminer.rule=Host(`db-admin.ovcloud.io`)
      - traefik.http.routers.adminer.tls=true
      - traefik.http.routers.adminer.tls.certresolver=lets-encrypt
      - traefik.port=8080
    networks:
      - internal
      - web
    depends_on:
      - db

volumes:
  data: {}
  cellsdir: {}
  mysqldir: {}
